access-common/src/main/java/edu/unc/lib/dl/ui/util/StringFormatUtil.java

```diff
@@ -17,84 +17,155 @@
-                    decimalMultiplier *= 10;
-                }
-                filesizeRemainder = (filesizeRemainder * decimalMultiplier) / 1024;
-                if (filesizeRemainder > 0) {
-                    filesizeBuilder.append(".").append(filesizeRemainder);
-                }
+            filesizeRemainder = (filesizeRemainder * decimalMultiplier) / 1024;
+            if (filesizeRemainder > 0) {
+                filesizeBuilder.append(".").append(filesizeRemainder);
             }
+        }
 
-            filesizeBuilder.append(" ");
-            if (multipleCount < filesizeSuffixes.length) {
-                filesizeBuilder.append(filesizeSuffixes[multipleCount]);
-            } else {
-                filesizeBuilder.append(filesizeSuffixes[filesizeSuffixes.length - 1]);
-            }
-            return filesizeBuilder.toString();
-        } catch (NumberFormatException e) {
+        filesizeBuilder.append(" ");
+        if (multipleCount < filesizeSuffixes.length) {
+            filesizeBuilder.append(filesizeSuffixes[multipleCount]);
+        } else {
+            filesizeBuilder.append(filesizeSuffixes[filesizeSuffixes.length - 1]);
         }
-        return filesize;
+        return filesizeBuilder.toString();
     }
 
+    /**
+     * Truncates the given text to no more than the requested length, in a human
+     * readable manner if possible.
+     *
+     * If this would cut off in the middle of a word, will attempt to truncate
+     * to before this word within reason.
+     *
+     * @param text string to truncate
+     * @param length maximum length of the truncated text
+     * @return text truncated to be no more than the requested length.
+     */
     public static String truncateText(String text, int length) {
+        return truncateText(text, length, DEFAULT_LAST_WORD_ALLOWANCE);
+    }
+
+    private static String truncateText(String text, int length, int lastWordAllowance) {
         if (length < 0) {
             throw new IndexOutOfBoundsException();
         }
 
         if (text == null || text.length() <= length) {
             return text;
         }
-        Matcher m = Pattern.compile("(.|\n) {0," + length + "}\\b").matcher(text);
-        m.find();
-        return m.group(0);
+
+        String truncated;
+        if (length < text.length()) {
+            truncated = text.substring(0, length);
+        } else {
+            truncated = text;
+        }
+
+        // Truncate text down to last word boundry if it occurs within
+        // lastWordAllowance number of characters from the end of the truncated
+        // text. Effectively this will try to cut off to the last whole word in
+        // the text within the character limit
+        if (truncated.length() > lastWordAllowance * LAST_WORD_ALLOWANCE_MIN_LENGTH_FACTOR) {
+            String[] parts = truncated.split("\\b(?=\\w*$)");
+            if (parts.length > 1) {
+                int lastBoundryIndex = parts[0].length();
```

Comment:
"boundaryIndex"

Reply:
Its actually the index of the boundry for the last word, so I don't think the name is inaccurate?